engine/src/main/java/io/camunda/zeebe/process/test/engine/IdleStateMonitor.java

```diff
@@ -44,9 +45,16 @@
     reader = logStreamReader;
   }
 
-  public void addCallback(final Runnable callback) {
-    synchronized (callbacks) {
-      callbacks.add(callback);
+  public void addOnIdleCallback(final Runnable callback) {
+    synchronized (idleCallbacks) {
+      idleCallbacks.add(callback);
+    }
+    checkIdleState();
```

Comment:
:x: rename method to something like `checkEngineState` or `checkProcessingOrIdleState` or something along those lines.

Maybe we should also rename the method to `checkStateAndNotifyCallbacks` to hint to the side effect

Reply:
:x: dito for renaming the class altogether